File: INTRODUCCION-POO/EJERCICIO-7/Celular.py
class Celular:
    def __init__(self):
        self.espacio = 1024
        self.bateria = 100
        self.cantidadApp = 0

    def nuevaApp(self, nombre, tamaño):
        self.nombre = nombre
        self.tamaño = tamaño
        self.cantidadApp +=1
        self.espacio -= self.tamaño
        if self.cantidadApp > 20:
            return print("Espacio de aplicaciones llenas")
        elif self.espacio < 1:
            return print("Espacio lleno!!!")
        else:
            return print(f"{self.nombre} instalado espacio restante {self.espacio}")
    def App(self):
        if self.tamaño > 100 and self.tamaño <250:
            self.bateria -= 2
        elif self.tamaño > 250:
            self.bateria -= 5
        else:
            self.bateria -= 1
        if self.bateria < 1:
            self.bateria = 0
            return print("celular apagado") 
        return f"bateria: {self.bateria}"
    def mostrarBateria(self):
        return f"BATERIA: {self.bateria}"

x = Celular()
x.nuevaApp("pol",254)
#por cada app es el uso por 10 minutos 
x.App()
x.App()
x.App()
x.App()
print(x.mostrarBateria())
x.nuevaApp("uui",23)
x.App()
print(x.mostrarBateria())
x.nuevaApp("face",144)
x.App()
print(x.mostrarBateria())

File: INTRODUCCION-POO/EJERCICIO-7/Celular.java
package ejer;

class Celular {
	int espacio = 1024;
	int bateria = 100;
	int cantidadApp = 0;
	public void nuevaApp(String nombre, int tamaño) {
		cantidadApp += 1;
		espacio -= tamaño;
		if(cantidadApp > 20) {
			System.err.println("Espacio de aplicaciones llenas!!");
		} if(espacio < 1) {
			System.err.println("Espacio lleno!!!");
		}else {
			System.out.println(nombre + " instalado. espacio disponible: "+espacio);
		}
		return;
	}
	public void App(String nombre,int tamaño) {
		if (tamaño > 100 && tamaño < 250) {
			bateria -=2;
		} if(tamaño > 250) {
			bateria -=5;
		} else {
			bateria -=1;
		}
		if (bateria < 1) {
			bateria = 0;
			System.err.println("CELULAR APAGADO");
		}
		return;
	}
	public void mostrarBateria() {
		System.out.println("Bateria: "+bateria);
		return;
	}
	public static void main(String[] args) {
		Celular x = new Celular();
		x.nuevaApp("foco", 295);
		x.App("foco",295);
		x.mostrarBateria();
		x.App("foco",295);
		x.mostrarBateria();
		x.App("foco",295);
		x.mostrarBateria();
		x.nuevaApp("wes", 79);
		x.App("wes",79);
		x.mostrarBateria();
		x.nuevaApp("qys", 159);
		x.App("qys",159);
		x.mostrarBateria();
		
	}
}
